iroh/src/discovery/local_swarm_discovery.rs

```diff
@@ -348,7 +355,16 @@ fn peer_to_discovery_item(peer: &Peer, node_id: &NodeId) -> DiscoveryItem {
         .iter()
         .map(|(ip, port)| SocketAddr::new(*ip, *port))
         .collect();
-    let node_info = NodeInfo::new(*node_id).with_direct_addresses(direct_addresses);
+    // Get the user-defined data from the resolved peer info. We expect an attribute with a value
+    // that parses as `UserData`. Otherwise, omit.
+    let user_data = if let Some(Some(user_data)) = peer.txt_attribute(USER_DATA_ATTRIBUTE) {
+        user_data.parse().ok()
```

Comment:
I think we should print at least a debug message of the errror here, so that it isn't fully silent if there is invalid data floating around, and you are trying to debug why it can't get resolved